src/foamadapter/solver/icofoam.py

```diff
@@ -0,0 +1,101 @@
+import pybFoam
+from pybFoam import (
+    Info,
+    Time,
+    Word,
+    adjustPhi,
+    constrainHbyA,
+    constrainPressure,
+    createPhi,
+    dictionary,
+    fvc,
+    fvm,
+    fvMesh,
+    fvScalarMatrix,
+    fvVectorMatrix,
+    pisoControl,
+    setRefCell,
+    solve,
+    surfaceScalarField,
+    volScalarField,
+    volVectorField,
+)
+
+
+def create_fields(mesh):
+    p = volScalarField.read_field(mesh, "p")
+    U = volVectorField.read_field(mesh, "U")
+    phi = createPhi(U)
+    nu = volScalarField.read_field(mesh, "nu")  # Assumes viscosity is read like a field
+
+    return p, U, phi, nu
+
+
+class IcoFoam:
+    def __init__(self, argv):
+        self._argv = argv
+
+    def run(self):
+        argList = pybFoam.argList(self._argv)
+
+        runTime = Time(argList)
+
+        mesh = fvMesh(runTime)
+
+        p, U, phi, nu = create_fields(mesh)
+
+        # Optional: pressure reference
+
+        fvSolution = dictionary.read("system/fvSolution")
+
+        pRefCell, pRefValue = setRefCell(p, fvSolution.subDict("PISO"))
+
+        mesh.setFluxRequired(Word("p"))
```

Comment:
Doesn't this depend on flags in fvSchemes?
```suggestion
        mesh.setFluxRequired(fvSchemes)
```

Reply:
This is how it is implemented in OpenFOAM ...